source/CompactRegular.lagda

```diff
@@ -793,3 +793,187 @@ clopen-iff-compact-in-stone-frame F (κ , ζ) U = β , γ
+ where
+  γ : is-spectral₀ F → ∥ Σ ℬ ꞉ Fam _ ⟨ F ⟩ , is-basis-for F ℬ ∥
+  γ (ℬ , p) = ∣ ℬ , pr₁ p ∣
+
+\end{code}
+
+\begin{code}
+
+cofinal-in : (F : frame 𝓤 𝓥 𝓦) → Fam 𝓦 ⟨ F ⟩ → Fam 𝓦 ⟨ F ⟩ → Ω (𝓥 ⊔ 𝓦)
+cofinal-in F R S =
+ Ɐ i ∶ index R , Ǝ j ∶ index S , ((R [ i ]) ≤[ poset-of F ] (S [ j ])) holds
+
+cofinal-implies-join-covered : (F : frame 𝓤 𝓥 𝓦) (R S : Fam 𝓦 ⟨ F ⟩)
+                             → cofinal-in F R S holds
+                             → ((⋁[ F ] R) ≤[ poset-of F ] (⋁[ F ] S)) holds
+cofinal-implies-join-covered F R S φ = ⋁[ F ]-least R ((⋁[ F ] S) , β)
+ where
+  open PosetReasoning (poset-of F)
+
+  β : (i : index R) → ((R [ i ]) ≤[ poset-of F ] (⋁[ F ] S)) holds
+  β i = ∥∥-rec (holds-is-prop ((R [ i ]) ≤[ poset-of F ] (⋁[ F ] S))) γ (φ i)
+   where
+    γ : Σ j ꞉ index S , ((R [ i ]) ≤[ poset-of F ] (S [ j ])) holds
+        → ((R [ i ]) ≤[ poset-of F ] (⋁[ F ] S)) holds
+    γ (j , p) = R [ i ] ≤⟨ p ⟩ S [ j ] ≤⟨ ⋁[ F ]-upper S j ⟩ ⋁[ F ] S ■
+
+bicofinal-implies-same-join : (F : frame 𝓤 𝓥 𝓦) (R S : Fam 𝓦 ⟨ F ⟩)
+                            → cofinal-in F R S holds
+                            → cofinal-in F S R holds
+                            → ⋁[ F ] R ≡ ⋁[ F ] S
+bicofinal-implies-same-join F R S φ ψ =
+ ≤-is-antisymmetric
+  (poset-of F)
+  (cofinal-implies-join-covered F R S φ)
+  (cofinal-implies-join-covered F S R ψ)
+
+compact-rel-syntax : (F : frame 𝓤 𝓥 𝓦) → ⟨ F ⟩ → ⟨ F ⟩ → Ω (𝓤 ⊔ 𝓥 ⊔ 𝓦 ⁺)
+compact-rel-syntax F U V =
+ Ɐ W ∶ ⟨ F ⟩ , is-compact-open F W ⇒ W ≤[ poset-of F ] U ⇒ W ≤[ poset-of F ] V
+
+syntax compact-rel-syntax F U V = U ≤ₖ[ F ] V
+
+spectral-yoneda : (F : frame 𝓤 𝓥 𝓦) → is-spectral F holds → (U V : ⟨ F ⟩)
+                → (U ≤ₖ[ F ] V ⇒ U ≤[ poset-of F ] V) holds
+spectral-yoneda {𝓦 = 𝓦} F σ U V χ =
+ ∥∥-rec (holds-is-prop (U ≤[ poset-of F ] V)) γ σ
+  where
+   open PosetReasoning (poset-of F)
+   open Joins (λ x y → x ≤[ poset-of F ] y)
+   open JoinNotation (λ - → ⋁[ F ] -)
+
+   γ : is-spectral₀ F → (U ≤[ poset-of F ] V) holds
+   γ (ℬ , υ , φ , ψ) =
+    U                            ≡⟨ I  ⟩ₚ
+    ⋁[ F ] ⁅ ℬ [ i ] ∣ i ε ℐ ⁆   ≤⟨ ii ⟩
+    V                            ■
+    where
+     ℐ : Fam 𝓦 (index ℬ)
+     ℐ = pr₁ (υ U)
+
+     I : U ≡ ⋁[ F ] ⁅ ℬ [ i ] ∣ i ε ℐ ⁆
+     I = ⋁[ F ]-unique ⁅ ℬ [ i ] ∣ i ε ℐ ⁆ U (pr₂ (υ U))
+
+     ϑ : (i : index ℐ) → ((ℬ [ ℐ [ i ] ]) ≤[ poset-of F ] U) holds
+     ϑ i = ℬ [ ℐ [ i ] ]               ≤⟨ ⋁[ F ]-upper ⁅ ℬ [ i ] ∣ i ε ℐ ⁆ i ⟩
+           ⋁[ F ] ⁅ ℬ [ i ] ∣ i ε ℐ ⁆  ≡⟨ I ⁻¹                               ⟩ₚ
+           U                           ■
+
+     ξ : (V is-an-upper-bound-of ⁅ ℬ [ i ] ∣ i ε ℐ ⁆) holds
+     ξ i = χ (ℬ [ ℐ [ i ] ]) (φ (ℐ [ i ])) (ϑ i)
+
+     ii : ((⋁[ F ] ⁅ ℬ [ i ] ∣ i ε ℐ ⁆) ≤[ poset-of F ] V) holds
+     ii = ⋁[ F ]-least ⁅ ℬ [ i ] ∣ i ε ℐ ⁆ (V , ξ)
+
+\end{code}
+
+\begin{code}
+
+compacts-are-basic-in-spectral-frames : (F : frame 𝓤 𝓥 𝓦)
+                                      → (σ : is-spectral₀ F)
```

Comment:
Rename "is-spectral₀" to "spectral-structure".

Reply:
Or "spectral-data" maybe is better.

Reply:
Rename "compacts-are-basic-in-spectral-frames" to "compacts-are-basic-in-spectralᴰ-frames".

Whenever you work with spectrality as data, use a "D" superscript as above.

Reply:
More generally, whenever you work with the data version of something, use a "D" superscript.